web3/_utils/filters.py

```diff
@@ -93,30 +122,19 @@ def construct_event_filter_params(
     filter_params["topics"] = topic_set
 
     if address and contract_address:
+        filter_params["address"] = _validate_addresses_as_list(
+            address
+        ) + _validate_addresses_as_list(contract_address)
+    elif address and not contract_address:
         if is_list_like(address):
-            filter_params["address"] = [address] + [contract_address]
-        elif is_string(address):
-            filter_params["address"] = (
-                [address, contract_address]
-                if address != contract_address
```

Comment:
It looks like we're missing this check. Do we need it?

Reply:
If I'm reading it right, it looks like we would have two of the same address if they are the same, so might be good to write a test for that case if we don't already have one... and / or check that the address list formed is comprised of unique addresses?

Reply:
It does look like we're missing the test case where they are the same. But going a step further, what happens if one is inside a list and the other isn't and they are the same address? I don't think we really look in the list we just compare `a != b`. If we're being thorough, we should have that check as well.

We can maybe leverage sets here to make sure they are unique before converting to lists?

Reply:
@fselmo My reasoning here was that the rpc call can contain duplicates and clients make the list a set as well. Since it's no cost to deduplicate them before sending the request, I will update this behavior.

As for the checksums, I will add that as part of the validation checks.

Thanks for the feedback!

Reply:
I see. I'm indifferent. It would maybe be better to clean up any duplicates but if we can guarantee somewhere in the spec that duplicates are fine and leads to expected behavior then I'm good with keeping this simpler!